course05/homework01/harrovanderkroft/string.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
```

Comment:
In general:

I would put the definitions into a cc file.
Check the Opt coding guidelines for bracket style, unfortunately it requires East coast style (or however it is called) over K&R.
There are also some wrong indentations.